src/commands/attest/create.ts

```diff
@@ -0,0 +1,136 @@
+import {Command, Flags} from '@oclif/core'
+import {Attestation} from '@mitre/hdf-converters'
+import fs from 'fs'
+import AccurateSearch from 'accurate-search'
+import XlsxPopulate from 'xlsx-populate'
+import {ExecJSON} from 'inspecjs'
+import promptSync from 'prompt-sync'
+import {default as files} from '../../resources/files.json'
+import {dataURLtoU8Array} from '../../utils/global'
+import yaml from 'yaml'
+
+const MAX_SEARCH_RESULTS = 5
+const prompt = promptSync()
+
+export default class CreateAttestations extends Command {
+    static flags = {
+      help: Flags.help({char: 'h'}),
+      input: Flags.string({char: 'i', description: '(optional) An input HDF file to search for controls'}),
+      output: Flags.string({char: 'o', required: true, description: 'The output filename'}),
+      format: Flags.string({char: 't', description: '(optional) The output file type', default: 'json', options: ['json', 'xlsx', 'yml', 'yaml']}),
+    }
+
+    promptForever(promptValue: string): string {
+      while (true) {
+        const ret = prompt(promptValue)
+        if (ret.trim() !== '') {
+          return ret
+        }
+      }
+    }
+
+    getStatus(): 'passed' | 'failed' {
+      const validPassResponses = new Set(['p', 'passed', 'pass'])
+      const validFailResponses = new Set(['f', 'failed', 'fail', 'failure'])
+      while (true) {
+        const input = prompt('Enter status ((p)assed/(f)ailed): ') || ''
+        if (validPassResponses.has(input.trim().toLowerCase())) {
+          return 'passed'
+        }
+
+        if (validFailResponses.has(input.trim().toLowerCase())) {
+          return 'failed'
+        }
+      }
+    }
+
+    promptForAttestation(id: string): Attestation {
+      return {
+        control_id: id,
+        explanation: this.promptForever('Attestation explanation: '),
+        frequency: this.promptForever('Frequency (1d/3d/1wk/2wk/1m/3m/6m/1y/1.5y/custom): '),
+        status: this.getStatus(),
+        updated: new Date().toISOString(),
+        updated_by: this.promptForever('Updated By: '),
+      }
+    }
+
+    async run() {
+      const {flags} = await this.parse(CreateAttestations)
+
+      const attestations: Attestation[] = []
+
+      if (flags.input) {
+        const evaluation = JSON.parse(fs.readFileSync(flags.input, 'utf8')) as ExecJSON.Execution
+        const search = new AccurateSearch()
+        const controls: Record<string, ExecJSON.Control> = {}
+        for (const profile of evaluation.profiles) {
+          for (const control of profile.controls) {
+            controls[control.id] = control
+            search.addText(control.id, control.id + ': ' + (control.title || '') + ' ' + control.desc || '')
```

Comment:
shouldn't contro.desc || '' also be in parentheses?

Reply:
Because it's at the end of the function call it works without this

Reply:
I disagree: 

```
> a
{ a: 'a', b: 'b' }
> a.a + ": " + (a.c || '') + " " + a.d || ''
'a:  undefined'
```

That control.desc could undefined or null and consequently would show up like that in the string.  This is cause logical OR is way below addition in terms of operator precedence.

Tangent: that generated string should probs also be a template string anyways